src/js/providers/html5.js

```diff
@@ -476,9 +476,16 @@ function VideoProvider(_playerId, _playerConfig, mediaElement) {
                     currentQuality = i;
                 }
                 if (label && levels[i].label === label) {
-                    return i;
+                    currentQuality = i;
+                    break;
                 }
             }
+            if (!currentQuality) {
+                // Choose the highest quality
+                const sorted = levels.map((level, index) => Object.assign({ index, height: 0 }, level))
+                    .sort((a, b) => b.height - a.height);
+                currentQuality = sorted[0].index;
+            }
         }
         visualQuality.reason = 'initial choice';
```

Comment:
is it safe to fall through and hit this block? previously the reason would never get set if we found the level above

Reply:
Yeah. That was a mistake.